src/core/Database.cpp

```diff
@@ -308,6 +308,22 @@ void Database::recycleGroup(Group* group)
      }
 }
 
+void Database::emptyRecycleBin()
+{
+    if (m_metadata->recycleBinEnabled() && m_metadata->recycleBin()) {
+        // destroying direct entries of the recycle bin
+        QList<Entry*> subEntries = m_metadata->recycleBin()->entries();
+        for (Entry* entry : subEntries) {
+            delete entry;
```

Comment:
@droidmonkey Well, it seems it is, or almost it is. Here is the code from Group destructor ([src link](https://github.com/keepassxreboot/keepassxc/blob/develop/src/core/Group.cpp)):
```c++
Group::~Group()
{
    // Destroy entries and children manually so DeletedObjects can be added
    // to database.
    const QList<Entry*> entries = m_entries;
    for (Entry* entry : entries) {
        delete entry;
    }

    const QList<Group*> children = m_children;
    for (Group* group : children) {
        delete group;
    }

    if (m_db && m_parent) {
        DeletedObject delGroup;
        delGroup.deletionTime = QDateTime::currentDateTimeUtc();
        delGroup.uuid = m_uuid;
        m_db->addDeletedObject(delGroup);
    }

    cleanupParent();
}
```
As you can see, it does things similar to what I've done.